arelle/plugin/validate/ROS/rules/ros.py

```diff
@@ -0,0 +1,321 @@
+                                    for f in fList[1:]:
+                                        _d = inferredDecimals(f)
+                                        _v = f.xValue
+                                        if isnan(_v):
+                                            _inConsistent = True
+                                            break
+                                        if _d in decVals:
+                                            _inConsistent |= _v != decVals[_d]
+                                        else:
+                                            decVals[_d] = _v
+                                        a, b, _inclA, _inclB = rangeValue(_v, _d)
+                                        if a > aMax: aMax = a
+                                        if b < bMin: bMin = b
+                                    if not _inConsistent:
+                                        _inConsistent = (bMin < aMax)
+                                    decVals.clear()
+                            else: # string complete duplicates
+                                _inConsistent = any(not f.isVEqualTo(f0) for f in fList[1:])
+                            if _inConsistent:
+                                modelXbrl.error("ROS.inconsistentDuplicateFacts",
+                                                "Inconsistent duplicate fact values %(element)s: %(values)s, in contexts: %(contextIDs)s.",
+                                                modelObject=fList, element=f0.qname,
+                                                contextIDs=", ".join(sorted(set(f.contextID for f in fList))),
+                                                values=", ".join(strTruncate(f.value,64) for f in fList))
+                aspectEqualFacts.clear()
+        del factForConceptContextUnitHash, aspectEqualFacts
+    modelXbrl.profileActivity(_statusMsg, minTimeToShow=0.0)
+    modelXbrl.modelManager.showStatus(None)
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_ros6(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    ROS: Rule 6: ProfitLossOnOrdinaryActivitiesBeforeTax (for IE GAAP)
+    OR ProfitLossBeforeTax (for IE IFRS) must exist in the document and be non-nil.
+    """
+    modelXbrl = val.modelXbrl
+    conceptLn = None
+    for filingType in pluginData.getFilingTypes(modelXbrl):
+        if NAMESPACE_IE_FRS_101 in filingType or NAMESPACE_IE_FRS_102 in filingType:
+            conceptLn = IE_GAAP_PROFIT_LOSS
+        elif NAMESPACE_IE_IFRS in NAMESPACE_IE_IFRS:
+            conceptLn = IE_IFRS_PROFIT_LOSS
+    if conceptLn:
+        return errorOnRequiredFact(
+            val.modelXbrl,
+            factLn=conceptLn,
+            code='ROS.6',
+            message=_("Error code 6: ProfitLossOnOrdinaryActivitiesBeforeTax (for IE GAAP)"
```

Comment:
I would update this message so that only the concept that we care about gets returned rather than both concepts.